File: OP_PROJECT_JAVA/src/game/Animals/Human.java
package game.Animals;

import game.Animal;
import game.Position;
import game.World;

public class Human extends Animal {
    private static final int HUMAN_STRENGTH = 5;
    private static final int HUMAN_INIT = 4;
    private static final String HUMAN_TYPE = "Human";


    public Human(Position position, World world) {
        super(HUMAN_TYPE, HUMAN_STRENGTH, HUMAN_INIT, position, world);
        this.world = world;
    }

    @Override
    public void action() {
        if (world.getCooldown() == 10) {
            strength += 5;
        } else if (world.getCooldown() >= 5) {
            strength--;
        }

        char nextMove = world.getNextMove();
        Position newPosition = new Position(position.getY(), position.getX());
        switch (nextMove) {
            case 'w':
                newPosition.setY(newPosition.getY() - 1);
                break;
            case 's':
                newPosition.setY(newPosition.getY() + 1);
                break;
            case 'a':
                newPosition.setX(newPosition.getX() - 1);
                break;
            case 'd':
                newPosition.setX(newPosition.getX() + 1);
                break;
            default:
                break;
        }

        if (world.isPositionWithinBounds(newPosition)) {
            if (world.getOrganismAtPosition(newPosition) != null) {
                collision(world.getOrganismAtPosition(newPosition));
            } else {
                moveOrganism(newPosition);
            }
        }
    }

    public void setWorld(World world) {
        this.world = world;
    }

}

File: OP_PROJECT_PYTHON/src/game/Animals/Human.py
from src.game.Animal import Animal
from src.game.Position import Position


class Human(Animal):
    HUMAN_STRENGTH = 5
    HUMAN_INIT = 4
    HUMAN_TYPE = "Human"

    def __init__(self, position, world):
        super().__init__(Human.HUMAN_TYPE, Human.HUMAN_STRENGTH, Human.HUMAN_INIT, position, world)
        self.world = world

    def action(self):
        if self.world.cooldown == 10:
            self.strength += 5
        elif self.world.cooldown >= 5:
            self.strength -= 1

        next_move = self.world.next_move
        new_position = Position(self.position.get_y(), self.position.get_x())

        if next_move == 'w':
            new_position.set_y(new_position.get_y() - 1)
        elif next_move == 's':
            new_position.set_y(new_position.get_y() + 1)
        elif next_move == 'a':
            new_position.set_x(new_position.get_x() - 1)
        elif next_move == 'd':
            new_position.set_x(new_position.get_x() + 1)

        if self.world.is_position_within_bounds(new_position):
            organism_at_new_position = self.world.get_organism_at_position(new_position)
            if organism_at_new_position is not None:
                self.collision(organism_at_new_position)
            else:
                self.move(new_position)

    def set_world(self, world):
        self.world = world
